### DISCUSSION TITLE
Alpine "forgetting" attributes when moving dom elements around

### DISCUSSION THREAD

**Discussion by sergio-agosti** [alpinejs/alpine#4449] [7. Bugs (archived)] [answered]
On Alpine 3.14.4, when moving dom elements that have Alpine attributes attached, Alpine "forgets" those attributes. In the following examples, some elements with x-show attributes get moved inside the x-init event. With Alpine 3.14.3 everything works as expected.

Example:

Alpine 3.14.3: https://codepen.io/sergio-agosti/pen/YzmdeZO
Alpine 3.14.4: https://codepen.io/sergio-agosti/pen/raBNKGY

**Comment by ekwoka** [comment]
Definitely related to https://github.com/alpinejs/alpine/pull/4447

:thinking:

It's not so much forgetting the attributes, but that the elements initially are queued to initialize, and then get moved, and it's messing things up. This is a rather contrived case that doesn't make much sense, but I do agree that it SHOULD work.

**Reply by ekwoka** [reply]
Actually, yeah, it's losing them completely

https://awesomealpine.com/play?share=xDQx47SpDlnk

Here's a demo with a delay as well to show that it is actually removing applied things. Very curious.

**Reply by ekwoka** [reply]
hmm, in my own test version of Alpine, these changes don't cause this issue...very strange...

Okay, that's unrelated. The issue was happening, but my test of it occuring just didn't seem to be failing.

**Comment by ekwoka** [comment, marked as answer]
Fix is in #4450

**Reply by sergio-agosti** [reply]
Wow that was quick, thanks @ekwoka !

**Reply by ekwoka** [reply]
It was luckily something that identification was quick enough, and the solution even quicker
